server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -260,6 +260,13 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> CCS_COLLECT_TELEMETRY = Setting.boolSetting(
+        "search.ccs.collect_telemetry",
+        true,
+        Property.Dynamic,
```

Comment:
asking to learn - this means the user can change this setting via the Settings API and not do a restart - is that right?

Reply:
Yes, generally I don't see any reason not to allow it - it will stop collecting telemetry immediately then, which one can use if e.g. telemetry collection produces some undesired effects for some reason.